File: output/deepseek_chat/avatar/algo_based_translation/java/python/successful_translation/atcoder_ABC123_C.py
import math

def main():
    N = int(input())
    min_val = 0
    for i in range(5):
        if i == 0:
            min_val = int(input())
        else:
            num = int(input())
            min_val = min(min_val, num)
    result = math.ceil(N / min_val) + 4
    print(result)

if __name__ == "__main__":
    main()

File: output/gpt_4o/avatar/algo_based_translation/python/java/failed_translation/atcoder_ABC123_C.java

import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;
import java.lang.Math;

public class atcoder_ABC123_C {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int N = scanner.nextInt();
        List<Integer> A = new ArrayList<>();
        
        for (int i = 0; i < 5; i++) {
            A.add(scanner.nextInt());
        }
        
        int minA = Collections.min(A);
        int result = (int) Math.ceil((double) N / minA) + 4;
        
        System.out.println(result);
        
        scanner.close();
    }
}

